File: exercises/cpp/lab12/fork_sum.cpp
// Lab 12 - Child computes sum of odd numbers, Parent computes sum of even numbers
#include <iostream>
#include <sys/wait.h>
#include <unistd.h>
using namespace std;

int main() {
    int a[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int n = sizeof(a) / sizeof(a[0]);

    printf("=== Fork: Even/Odd Sum ===\n\n");

    pid_t pid = fork();

    if (pid < 0) {
        perror("fork");
        return 1;
    } else if (pid > 0) {
        // Parent: sum even numbers
        int sumEven = 0;
        for (int i = 0; i < n; i++)
            if (a[i] % 2 == 0) sumEven += a[i];
        cout << "Parent process => Sum of even numbers (1-10): " << sumEven << "\n";
        wait(NULL);
    } else {
        // Child: sum odd numbers
        int sumOdd = 0;
        for (int i = 0; i < n; i++)
            if (a[i] % 2 != 0) sumOdd += a[i];
        cout << "Child  process => Sum of odd numbers  (1-10): " << sumOdd  << "\n";
    }

    return 0;
}

File: exercises/python/lab12/fork_sum.py
"""Lab 12 - Child computes sum of odd numbers, Parent computes sum of even numbers"""
import os
import sys

def main():
    print("=== Fork: Even/Odd Sum ===\n")
    numbers = list(range(1, 11))

    pid = os.fork()

    if pid < 0:
        print("fork failed", file=sys.stderr)
        sys.exit(1)
    elif pid > 0:
        sum_even = sum(x for x in numbers if x % 2 == 0)
        print(f"Parent process => Sum of even numbers (1-10): {sum_even}")
        os.wait()
    else:
        sum_odd = sum(x for x in numbers if x % 2 != 0)
        print(f"Child  process => Sum of odd numbers  (1-10): {sum_odd}")


if __name__ == "__main__":
    main()
